### ISSUE TITLE
Video accelerator (HW device setting) is ignored by internal LAV video decoder

### ISSUE THREAD

**Issue by ArminiusTux** [clsid2/mpc-hc#81]
Dear clsid2,

let me thank you first for your continued work on this neat & sleek media player!
It is my premiere choice when it comes to playing UltraHD bluray discs on Windows.

My issue is kind of related to that use - picking a desired HW video accelerator within the LAV video decoder settings:

![internal-LAV_Video_Decoder](https://user-images.githubusercontent.com/44533868/54496023-9fad6980-48ea-11e9-9a12-70edf99d8cdb.PNG)
Switching the GPUs on the interal video filter is pointless - the auto setting is always used.

To remedy this I use the external LAV video filter (activated in the MPC settings) which accepts and uses my secondary GPU:
![external-LAV_Video_Decoder](https://user-images.githubusercontent.com/44533868/54496057-fb77f280-48ea-11e9-81c4-5945326f0a30.PNG)

Needless to say that I prefer my passive Geforce over the monstrous Radeon Fury because of the more sophisticated video decoding acceleration, particularly for HEVC and VP9 material.

To make a long story short, all that is needed for MPC-HC is to "obey and remember" the user-selected HW decoding device within its internal LAV video filter.

Last but not least this nuisance is not related to any particular MPC or LAV Filter version - this situation has been present for years.

Regards,
ArminiusTux

**Comment by ArminiusTux** [comment]
After reading the following suggestion in issue #74 

> For problems with the internal codecs, you should go here:
https://github.com/Nevcairiel/LAVFilters

maybe @Nevcairiel would like to take an aim on this.

**Comment by Nevcairiel** [comment]
MPC-HC takes over configuration of LAV Filters if its used as internal filters. It would need to ensure to save all options itself, and apparently its skipping this one.

**Comment by clsid2** [comment]
MPC-HC correctly saves that setting for me for internal decoder.

**Comment by ArminiusTux** [comment]
> MPC-HC correctly saves that setting **for me** for internal decoder.

Maybe the "problem" only arises on ansychronous multiGPU systems (=different chip manufacturers).

@people with onboard GPU desktops / notebooks (e.g. Notebook with Intel onboard & nVidia addon GPUs) can you test this, please?

**Comment by clsid2** [comment]
The setting is stored in this registry key:
HKEY_CURRENT_USER\Software\MPC-HC\MPC-HC\Internal Filters\LAVVideo\HWAccel

**Comment by ArminiusTux** [comment]
> The setting is stored in this registry key:
> HKEY_CURRENT_USER\Software\MPC-HC\MPC-HC\Internal Filters\LAVVideo\HWAccel

Hello clsid2 - thanks for pointing out the location of the honeycomb, here are my system's values:

> External filter settings:
> ```
> 
> [HKEY_CURRENT_USER\Software\LAV\Video\HWAccel]
> "HWAccel"=dword:00000005
> "h264"=dword:00000001
> "vc1"=dword:00000001
> "mpeg2"=dword:00000001
> "mpeg4"=dword:00000001
> "dvd"=dword:00000001
> "hevc"=dword:00000001
> "vp9"=dword:00000001
> "HWResFlags"=dword:00000007
> "HWDeintMode"=dword:00000000
> "HWDeintOutput"=dword:00000000
> "HWAccelDeviceDXVA2"=dword:ffffffff
> "HWAccelDeviceDXVA2Desc"=dword:00000000
> "HWAccelDeviceD3D11"=dword:00000001
> "HWAccelDeviceD3D11Desc"=dword:00001d01
> "HWAccelCUVIDXVA"=dword:00000001
> ```
> 
> Internal filter settings:
> 
> ```
> [HKEY_CURRENT_USER\Software\MPC-HC\MPC-HC\Internal Filters\LAVVideo\HWAccel]
> "HWAccel"=dword:00000005
> "h264"=dword:00000001
> "vc1"=dword:00000001
> "mpeg2"=dword:00000001
> "mpeg4"=dword:00000001
> "dvd"=dword:00000001
> "hevc"=dword:00000001
> "HWResFlags"=dword:00000007
> "HWDeintMode"=dword:00000002
> "HWDeintOutput"=dword:00000000
> "HWDeintHQ"=dword:00000001
> "vp9"=dword:00000001
> "HWAccelDeviceDXVA2"=dword:ffffffff
> "HWAccelDeviceDXVA2Desc"=dword:00000000
> "HWAccelDeviceD3D11"=dword:ffffffff
> "HWAccelDeviceD3D11Desc"=dword:ffffffff
> "HWAccelCUVIDXVA"=dword:00000001
> ```
> 

If I set the following INTERNAL values manually via regedit,
```
"HWAccelDeviceD3D11"=dword:00000001
"HWAccelDeviceD3D11Desc"=dword:00001d01
```
the desired GPU (Geforce GT 1030) is used on the next MPC-HC startup - **BUT ONLY for ONE VIDEO playback**.
After playing the next video or another programme restart the settings are back to automatic = undesired scenario.

**Comment by clsid2** [comment]
Found a typo. Should be fixed in 1.8.6.

**Comment by ArminiusTux** [comment]
> fixed in 1.8.6.

Yes indeed the fix WORKS, thank you!
